### ISSUE TITLE
bug: still getting missing avante_templates

### ISSUE THREAD

**Issue by tobenwarrior** [yetone/avante.nvim#544]
### Describe the bug

i have updated to the latest version and followed all the previous reports with this issue but I am still getting it.
![Screenshot 2024-09-05 172047](https://github.com/user-attachments/assets/de11dd81-9026-41e5-ae65-998f31d6e56c)

### To reproduce

i get this issue when i ask a question and press enter. not when opening avante

### Environment

lazy vim

**Comment by darrentiang** [comment]
i don’t know

**Comment by sica07** [comment]
I have the same problem. I've checked  the `build` folder and `avante_templates.so` and `avante_tokenizers.so` are present.
I'm on linux (arch linux) and I'm using packer for package installations

**Comment by andrew-t-james-core** [comment]
Had some luck in (Lazyvim) completely uninstalling and cleaning up plugin. Reinstalled works now.

**Comment by sqr** [comment]
I added `build = "make"` to my config file and was still getting the error.

Moved the config out of the directory, cleaned up lazyvim like suggested above, quit, moved the config back in, restarted and it was working fine again.

**Comment by aarnphm** [comment]
check if you have set `version=false`. chances are the lazy-lock still lock to a version that is not main

**Comment by lucas-Nicolas** [comment]
I'm getting the same bug on macos with LazyVim even when tryign to clean and move the plugin back in. I have also tried manually making in the install directory but I get the same message.

**Comment by therod** [comment]
Had the same problem on macos. Following these steps solved the issue for me: 

1. Made sure avante builds from source using the`build = "make BUILD_FROM_SOURCE=true"` option.
2. Removed the avante.nvim directory under `~/.local/share/nvim/lazy/avante.nvim` and re-installing solved the issue.
3. Re-installed and now it works.

**Comment by tobenwarrior** [comment]
tried all the above still cant work I think there is a problem with the make file not able to run (I have make installed)
![Uploading Screenshot 2024-09-06 230522.png…]()

**Comment by dmaspataud** [comment]
I had the same issue, what fixed it for me was to 

- Remove everything under `~/.local/share/nvim/lazy/avante.nvim`
- Start nvim, :Lazy, Sync
- Build manually:
```
cd ~/.local/share/nvim/lazy/avante.nvim
make
```
- Restart nvim, and retry

**Comment by snowphone** [comment]
I guess for `vim-plug` users, `  Plug 'snowphone/avante.nvim', { 'branch': 'main', 'do': { -> avante#build('source=true') } }` should be used by default.

**Comment by tobenwarrior** [comment]
manual build isnt working i have all the files I tried :Lazy sync as well 
![Screenshot 2024-09-07 030229](https://github.com/user-attachments/assets/283f5748-6421-469e-89eb-df5e4315dd87)
![Screenshot 2024-09-07 030340](https://github.com/user-attachments/assets/16a962ac-819e-4d18-a156-a69ed2be2b10)
back to this error again

**Comment by sebassdc** [comment]
> I had the same issue, what fixed it for me was to
> 
> * Delete `~/.local/share/nvim/lazy/avante.nvim`
> * Start nvim, `:Lazy` + sync
> * Build manually:
> 
> ```
> cd ~/.local/share/nvim/lazy/avante.nvim
> make
> ```
> 
> * Restart nvim, and retry

This did the trick for me in mac with Lazy

**Comment by tobenwarrior** [comment]
i dont know how but i just randomly tried the solutions above and somehow it worked thanks guys!

**Comment by imcquee** [comment]
Still running into this despite trying the above multiple times. MacOS, 

avante_templates.dylib
avante_tokenizers.dylib

Are in my build folder. Same setup in linux works fine with .so

**Comment by tkreuder** [comment]
> Still running into this despite trying the above multiple times. MacOS,
> 
> avante_templates.dylib avante_tokenizers.dylib
> 
> Are in my build folder. Same setup in linux works fine with .so
> 
> Also on intel mac, could very well be the reason
> 
> EDIT
> 
> I apologize, building from source does indeed work on intel mac. If anyone else struggles with the above, instead of running make just run
> 
> ```shell
> make BUILD_FROM_SOURCE=true
> ```
> 
> and it should build properly. Fans might go to 50000rpm but it should work

Thanks a lot for your hint!

I had to install rust to get rid of:
```
make BUILD_FROM_SOURCE=true

cargo build --release --features=luajit -p avante-tokenizers
make: cargo: No such file or directory
make: *** [luajit-tokenizers] Error 1
```

via: `brew install rust` which includes cargo.

then i got this:

make BUILD_FROM_SOURCE=true
```

cargo build --release --features=luajit -p avante-tokenizers
    Updating crates.io index
    Updating git repository `https://github.com/mlua-rs/mlua.git`
error: failed to get `mlua` as a dependency of package `avante-templates v0.1.0 (/Users/torbenkreuder/.local/share/nvim/lazy/avante.nvim/crates/avante-templates)`

Caused by:
  failed to load source for dependency `mlua`

Caused by:
  Unable to update https://github.com/mlua-rs/mlua.git?branch=main#1634c43f

Caused by:
  failed to clone into: /Users/torbenkreuder/.cargo/git/db/mlua-7a97be9956039e84

Caused by:
  failed to authenticate when downloading repository: git@github.com:mlua-rs/mlua.git

  * attempted ssh-agent authentication, but no usernames succeeded: `git`

  if the git CLI succeeds then `net.git-fetch-with-cli` may help here
  https://doc.rust-lang.org/cargo/reference/config.html#netgit-fetch-with-cli

Caused by:
  no authentication methods succeeded
make: *** [luajit-tokenizers] Error 101
```
which i could solve with creating:
`~/.cargo/config.toml` and the following content:
```

[net]
git-fetch-with-cli = true
```

**Comment by James4Ever0** [comment]
I have tried all methods above and still won't work. Wonder if downgrading would help.

My machine is a Raspberry Pi 5.

**Comment by James4Ever0** [comment]
If I run `:lua require("avante_templates")` under folder `~/.local/share/nvim/lazy/avante.nvim/build` it will show:

```
E5108: Error executing lua error loading module 'avante_templates' from file './avante_templates.so':
        /snap/core22/current/usr/lib/aarch64-linux-gnu/libm.so.6: version `GLIBC_2.38' not found (required by ./av
ante_templates.so)
stack traceback:
        [C]: at 0xaaaac9ab74c0
        [C]: in function 'require'
        [string ":lua"]:1: in main chunk
```

Seems I need to use a different version of `nvim`.

**Comment by James4Ever0** [comment]
Compiling `nvim` from source has solved the problem.

**Comment by zstreet87** [comment]
With the BUILD_FROM_SOURCE=true, I'm seeing issues with finding openssl header files even though I have libssl-dev. I've set the env vars that are suggested in the error message but still seeing the build error.

Strange thing is, I have this plugin working fine on my other machines.


![image](https://github.com/user-attachments/assets/1a7be3d3-51a0-4cf9-86f1-7a46f4ec92d0)

**Comment by aarnphm** [comment]
what version of rust are you using here?

**Comment by zstreet87** [comment]
➜ rustc --version
rustc 1.81.0 (eeb90cda1 2024-09-04)

**Comment by zstreet87** [comment]
I got around this issue. The openssl my system was using was from conda so I moved conda to lower priority in my PATH. Then I set the following:
```bash
export OPENSSL_LIB_DIR=/usr/lib/x86_64-linux-gnu
export OPENSSL_INCLUDE_DIR=/usr/include/openssl
export OPENSSL_STATIC=1
```
finally was able to rebuild from source successfully.

**Comment by cckate** [comment]
> Had the same problem on macos. Following these steps solved the issue for me:
> 
> 1. Made sure avante builds from source using the`build = "make BUILD_FROM_SOURCE=true"` option.
> 2. Removed the avante.nvim directory under `~/.local/share/nvim/lazy/avante.nvim` and re-installing solved the issue.
> 3. Re-installed and now it works.

Also had the same problem. this is really helpful, Rebuild avante solved my issue

**Comment by dmbhatti** [comment]
Not sure if this should have been closed without fix. Should work out of box. Please consider reopening.

**Comment by wilkinson4** [comment]
I also experienced this error on MacOS and vibe coded a workaround. Not sure if this needs to be addressed still, but here is the summary:

  What I found

   1. The normal make path on main appears broken because it looks for a release tag named release-v0.1.0, while the actual GitHub release is v0.1.0.
   2. The source-build path on macOS also appears broken: Rust builds libavante_templates.dylib (and the other native libs), but the install step expects/copys .so, so lua/avante_templates.so never gets created.

  Result 
  Avante installs, but Neovim fails at runtime with:

   Make sure to build avante (missing avante_templates)

  Workaround
  I was able to get it working by manually building the native crates and copying the macOS .dylib outputs into the filenames Neovim expects, e.g.:

   - libavante_templates.dylib -> lua/avante_templates.so
   - libavante_tokenizers.dylib -> lua/avante_tokenizers.so
   - libavante_repo_map.dylib -> lua/avante_repo_map.so
   - libavante_html2md.dylib -> lua/avante_html2md.so

  So this looks like an upstream macOS build/install issue rather than a user config problem.
